evm-contracts/src/v0.6/examples/VRFD20.sol

```diff
@@ -0,0 +1,164 @@
+
+    /**
+     * @notice Constructor inherits VRFConsumerBase
+     *
+     * @dev NETWORK: KOVAN
+     * @dev   Chainlink VRF Coordinator address: 0xdD3782915140c8f3b190B5D67eAc6dc5760C46E9
+     * @dev   LINK token address:                0xa36085F69e2889c224210F603D836748e7dC0088
+     * @dev   Key Hash:   0x6c3699283bda56ad74f6b855546325b68d482e983852a7a82979cc4807b641f4
+     * @dev   Fee:        0.1 LINK (100000000000000000)
+     *
+     * @param vrfCoordinator address of the VRF Coordinator
+     * @param link address of the LINK token
+     * @param keyHash bytes32 representing the hash of the VRF job
+     * @param fee uint256 fee to pay the VRF oracle
+     */
+    constructor(address vrfCoordinator, address link, bytes32 keyHash, uint256 fee)
+        public
+        VRFConsumerBase(vrfCoordinator, link)
+    {
+        s_keyHash = keyHash;
+        s_fee = fee;
+        
+    }
+
+    /**
+     * @notice Requests randomness from a user-provided seed
+     * @dev This is only an example implementation and not necessarily suitable for mainnet.
+     * @dev You must review your implementation details with extreme care.
+     *
+     * @param userProvidedSeed uint256 unpredictable seed
+     * @param roller address of the roller
+     */
+    function rollDice(uint256 userProvidedSeed, address roller) public onlyOwner returns (bytes32 requestId) {
+        require(LINK.balanceOf(address(this)) >= s_fee, "Not enough LINK to pay fee");
+        require(s_results[roller] == 0, "Already rolled");
+        requestId = requestRandomness(s_keyHash, s_fee, userProvidedSeed);
+        s_rollers[requestId] = roller;
+        s_results[roller] = ROLL_IN_PROGRESS;
+        emit DiceRolled(requestId, roller);
+    }
+
+    /**
+     * @notice Callback function used by VRF Coordinator to return the random number
+     * to this contract.
+     * @dev This is where you do something with randomness!
+     * @dev The VRF Coordinator will only send this function verified responses.
+     *
+     * @param requestId bytes32
+     * @param randomness The random result returned by the oracle
+     */
+    function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
+        uint256 result = randomness.mod(20).add(1);
+        s_results[s_rollers[requestId]] = result;
+        emit DiceLanded(requestId, result);
+    }
+
+    /**
+     * @notice Get the house assigned to the player once the address has rolled
+     * @param player address
+     * @return house as a string
+     */
+    function house(address player) public view returns (string memory) {
+        require(s_results[player] != 0, "Dice not rolled");
+        require(s_results[player] != ROLL_IN_PROGRESS, "Roll in progress");
+        return getHouseName(s_results[player]);
+    }
+
+    /**
+     * @notice Withdraw LINK from this contract.
+     * @param to the address to withdraw LINK to
+     * @param value the amount of LINK to withdraw
```

Comment:
I would add a note here that this is an example only, and in a real contract withdrawals should happen according to the [established withdrawal pattern](https://docs.soliditylang.org/en/v0.4.24/common-patterns.html#withdrawal-from-contracts), to protect users from burning their funds by fat-fingering the `to` address.